src/arrayvec.rs

```diff
@@ -579,6 +579,19 @@ impl<T, const CAP: usize> ArrayVec<T, CAP> {
         Ok(())
     }
 
+    /// Leak the contents of the vector and return a reference to them in a mutable slice.
+    ///
+    /// This sets len() to zero. The elements will not be dropped. The backing memory will
+    /// not be lost however, as the ArrayVec can be used again as soon as the returned slice
+    /// is dropped.
+    pub fn leak(&mut self) -> &mut [T] {
+        unsafe {
+            let s = slice::from_raw_parts_mut(self.as_mut_ptr(), self.len());
+            self.set_len(0);
```

Comment:
I would be more comfortable to swap the order of these two methods actually! 

A. We create a borrow of the whole vec as a &mut [T]
B. We modify self (set_len)

This is better in the order of B, then A, so that it's clear we don't invalidate the borrow. Miri apparently didn't react on this though?